barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/mock_transcript.hpp

```diff
@@ -0,0 +1,100 @@
+#include "barretenberg/ecc/curves/grumpkin/grumpkin.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
+#include <vector>
+namespace bb {
+/**
+ * @brief Mock transcript class used by IPA tests and fuzzer
+ *
+ * @details This transcript can send previously determined challenges instead of ones generated by Fiat Shamir. It can
+ * also store elements received from the prover
+ *
+ */
+class MockTranscript {
+  public:
+    // Vector of challenges sent to the verifier
+    std::vector<uint256_t> challenges;
+    // Vector of group elements received from the prover / sent to the verifier
+    std::vector<bb::curve::Grumpkin::AffineElement> group_elements;
+    // Vector of field elements received from the prover / sent to the verifier. uint256_t is used to ignore field type
+    std::vector<uint256_t> field_elements;
+
+    // Indices of the elements being sampled
+    size_t current_challenge_index = 0;
+    size_t current_field_index = 0;
+    size_t current_group_index = 0;
+
+    /**
+     * @brief Reset the transcript (requires to submit the challenges)
+     *
+     * @param challenges_ Challenges that will be sent to the prover/verifier
+     * @param group_elements_ Group elements sent to the verifier
+     * @param field_elements_ Field elements sent to the verifier
+     */
+    void reset(std::vector<uint256_t> challenges_,
+               std::vector<bb::curve::Grumpkin::AffineElement> group_elements_ = {},
+               std::vector<uint256_t> field_elements_ = {})
+    {
+        challenges = std::move(challenges_);
+        current_challenge_index = 0;
+        current_field_index = 0;
+        current_group_index = 0;
+        group_elements = std::move(group_elements_);
+        field_elements = std::move(field_elements_);
+    }
+    /**
+     * @brief Reset the indices of elements sampled after using the transcript with the prover
+     *
+     * @details After the transcipt received elements from the prover, this method allows to reset counters so that the
+     * verifier can receive those elements
+     */
+    void reset_for_verifier()
```

Comment:
doesn't reset already do these?

Reply:
nvm, saw the usage, i think it makes sense, maybe would name `reset_indices` or sth?

Reply:
Updated